### ISSUE TITLE
False positive in ScalaDocChecker?

### ISSUE THREAD

**Issue by aginiewicz** [scalastyle/scalastyle#123]
I'm trying to enable ScalaDocChecker for my project (using ScalaStyle 0.6.0). I'm getting something I think is false positive. In file:

```
package com.specdevs.specgine.animation

/** A common interface for all Tweenable objects. */
trait TweenInterface {

  /** Method used to update the desired value.
   *
   * @param dt Curent time.
   * @return flag that indicates whether the animation is finished or not.
   */
  def update(dt: Float): Boolean
}
```

It says:

```
Missing or badly formed ScalaDoc: Extra @param dt
```

in line 4, i.e. one where trait is declared. Am I missing something here? The documentation dt is for update method, and it clearly requires one. Other thing that worries me, is that for:

```
package com.specdevs.specgine.animation

/** Represents easing functions for [[com.specdevs.specgine.animation.Tween `Tween`]]s.
 *
 * @tparam T a type of interpolated value.
 */
trait Interpolator[+T] extends Function1[Float,T]
```

it says:

```
Missing or badly formed ScalaDoc: Missing
```

in line 7 (again, line with trait definition), so it is like it cannot see the doc at all.

Is there some way to make ScalaDocChecker work for above two cases, i.e. I misunderstand how I should format those docs (although I can generate html and it works as intended), or it is something that isn't implemented yet?

**Comment by sokrahta** [comment]
:+1: Would love to enable ScalaDocChecker but I'm getting 1-4 false positives per file.

**Comment by obcode** [comment]
Any news on this issue? I would love to use ScalaDocChecker but also getting some false positive.

**Comment by matthewfarwell** [comment]
I still haven't worked out why this is happening - I'm still looking at it.

**Comment by ajtkulov** [comment]
Smallest test that failed (in ScalaDocCheckerTest.scala)
```
  @Test def test5(): Unit = {
    val source =
      """package org.scalastyle.scalariform
        |
        |/**
        | * Doc
        | */
        |object X {
        |
        |}
      """.stripMargin

    assertErrors(List(), source)
  }
```

But this one will pass: 
```
  @Test def test6(): Unit = {
    val source =
      """
        |/**
        | * Doc
        | */
        |object X {
        |
        |}
      """.stripMargin

    assertErrors(List(), source)
  }
```

**Comment by ajtkulov** [comment]
On behalf of Artem Saitov, we find the bug/lack in scalariform. 

In https://github.com/daniel-trinh/scalariform in NewlineInferencer.class we should modify code as follows:
```scala
  private def translateToNewline(currentToken: Token): Token = {
    val currentHiddenTokens: HiddenTokens = currentToken.associatedWhitespaceAndComments
    val spannedTokens = currentHiddenTokens.span(!_.token.isScalaDocComment)
    val whiteSpaceTokens = HiddenTokens(spannedTokens._1.toList)
    val otherTokens = HiddenTokens(spannedTokens._2.toList)
    val rawText = delegate.text.substring(whiteSpaceTokens.offset, whiteSpaceTokens.lastCharacterOffset + 1)
    val text = if (whiteSpaceTokens.containsUnicodeEscape) whiteSpaceTokens.text else rawText
    val tokenType = if (containsBlankLine(text)) NEWLINES else NEWLINE
    val newlineToken = Token(tokenType, text, whiteSpaceTokens.offset, rawText)
    currentToken.associatedWhitespaceAndComments_ = otherTokens
    newlineToken.associatedWhitespaceAndComments_ = whiteSpaceTokens
    tokenToEmitNextTime = currentToken
    newlineToken
  }
```

It solves problems with package/import problem with ScalaDoc false positive in Scalastyle.
And doesn't break anything in scalariform/scalastyle.

I'm going to fork/pull request it in scalariform. Apparently, it touches (recompile/publish) scalariform/scalastyle/scalastyle-sbt-plugin as well.

**Comment by miguel0afd** [comment]
Hi,
Any news on this one?

Thanks in advance

**Comment by dvaston** [comment]
I have tested on ScalaStyle version 0.7.0 but I notice that this issue with false positives still happens. Is there any ETA when it might be fixed?
Thanks :)

**Comment by nemccarthy** [comment]
We are hitting this too? Any idea when the next release will be? and the sbt plugin?

**Comment by ajtkulov** [comment]
@dvaston @nemccarthy Could you provide some examples/test cases. 

As I understand https://github.com/scalastyle/scalastyle/pull/155 covers bug fix "package" in file header.

**Comment by dvaston** [comment]
@ajtkulov - thanks, will have a look at #155

**Comment by matthewfarwell** [comment]
I've just released 0.8.0 which contains #155.

**Comment by dvaston** [comment]
@matthewfarwell - thanks :)

**Comment by aginiewicz** [comment]
@matthewfarwell Thanks! I will check it soon. Any simple way to turn off this rule for single project only (testing)? I could appent "Test" or "Spec" to all class names, but it will take some time before I can actually see and confirm it is fixed.

**Comment by matthewfarwell** [comment]
Closing, presumed fixed. If it isn't, please reopen, or better, raise a new issue.
